src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -133,7 +133,7 @@ export class Selection {
             return;
         }
         this._currentSingle = value;
-        if (this._items.trackBy && value) {
+        if (this._items.all && this._items.trackBy && value) {
```

Comment:
This is just a question that came to me after looking at it again. Could it be a possibility that `this._items.all` was undefined because of asynchronous race condition. It wasn't yet populated that's why it's undefined?

Reply:
You can take a look at this issue for replication: https://github.com/vmware/clarity/issues/2002
It happens when the datagrid is empty but the selected item is still populated. Its a weird corner case.